modules/@apostrophecms/ui/ui/apos/mixins/AposArchiveMixin.js

```diff
@@ -12,22 +12,47 @@ export default {
     async archive(doc) {
       try {
         const moduleOptions = window.apos.modules[doc.type];
+        // Make sure that if there are any modified descendants we know about them
         const isPage = doc.slug.startsWith('/');
+        if (isPage) {
+          doc = await apos.http.get(`${moduleOptions.action}/${doc._id}`, {
+            draft: true,
+            qs: {
+              children: {
+                depth: 10
+              }
+            }
+          });
+        }
+        const isModified = findModified(doc);
+        const descendants = countDescendants(doc);
+        const draftDescendants = countDraftDescendants(doc);
         const action = window.apos.modules[doc.type].action;
         const isPublished = !!doc.lastPublishedAt;
         const isCurrentContext = doc.aposDocId === window.apos.adminBar.context.aposDocId;
-        const hasChildren = isPage && doc._children && doc._children.length;
         const plainType = isPage ? 'page' : (moduleOptions.label || 'content');
         let description = `You are going to archive the ${plainType} "${doc.title}"`;
 
-        if (hasChildren) {
-          description += `, which has ${doc._children.length} child ${plainType}${doc._children.length > 1 ? 's' : ''}`;
+        if (descendants > 0) {
+          description += `, which has ${descendants} child ${plainType}${descendants > 1 ? 's' : ''}`;
```

Comment:
We should look for the pluralLabel as well. We ask people to provide it, so we should use it for stuff like this.